ui/app/serializers/secret-engine.js

```diff
@@ -1,7 +1,23 @@
 import { assign } from '@ember/polyfills';
 import ApplicationSerializer from './application';
+import { EmbeddedRecordsMixin } from '@ember-data/serializer/rest';
+
+export default ApplicationSerializer.extend(EmbeddedRecordsMixin, {
+  attrs: {
+    config: { embedded: 'always' },
+  },
+
+  normalize(modelClass, data) {
+    // embedded records need a unique value to be stored
+    // set id for config to uuid of secret engine
+    if (data.config && !data.config.id) {
+      data.config.id = data.uuid;
+    }
+    // move version out of options so it can be defined on secret-engine model
```

Comment:
Why is this still in options, and then need to be moved back below? I couldn't find anything in the docs - just curious!

Reply:
Maybe that's not the best phrasing :upside_down_face:. In the `normalize` method of that serializer I set `data.version` to `data.options.version` so that I could move the attribute to the `secret-engine` model (it was formely a fragment with just that one attribute). Since the backend expects the `version` property to be in the `options` object I copy it back.

Reply:
Okay - that's what I was wondering - how the backend was expecting it! I noticed the change on the model (which seems like a nice cleanup for removing the fragment) But I hadn't seen any documentation about the backend needing it within `options` - thank you!